subsys/nonsecure/Kconfig

```diff
@@ -0,0 +1,576 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if BUILD_WITH_TFM || IS_SPM
+menu "Trusted Execution Non-Secure"
+
+menu "Peripheral Secure mapping"
+
+config NRF_FPU_SECURE
+	bool "FPU configured as secure peripheral"
+	depends on CPU_HAS_FPU
+	help
+	  FPU is implemented with user-selectable security mapping. If it has
+	  been configured as a secure peripheral then only the secure processing
+	  environment is able to access the peripheral.
+
+config NRF_DCNF_SECURE
+	bool "DCNF configured as secure peripheral"
+	depends on HAS_HW_NRF_DCNF
+	help
+	  DCNF is implemented with user-selectable security mapping. If it has
+	  been configured as a secure peripheral then only the secure processing
+	  environment is able to access the peripheral.
+
+config NRF_OSCILLATORS_SECURE
+	bool "OSCILLATORS configured as secure peripheral"
+	depends on HAS_HW_NRF_OSCILLATORS
+	help
+	  OSCILLATORS is implemented with user-selectable security mapping. If it has
+	  been configured as a secure peripheral then only the secure processing
+	  environment is able to access the peripheral.
+
+config NRF_REGULATORS_SECURE
+	bool "REGULATORS configured as secure peripheral"
+	depends on HAS_HW_NRF_REGULATORS
+	help
+	  REGULATORS is implemented with user-selectable security mapping. If it has
+	  been configured as a secure peripheral then only the secure processing
+	  environment is able to access the peripheral.
```

Comment:
Most of these look identical except the name of the peripheral, could we use a template here?

```
peripheral=REGULATORS
rsource "Kconfig.template.peripheral
```

```
# Kconfig.template.peripheral
config NRF_$(peripheral)_SECURE
    bool "$(peripheral) configured as secure peripheral"
    depends on HAS_HW_NRF_$(peripheral)
    help
      $(peripheral) is implemented with user ...
  ```

Reply:
Good suggestion, thank you.

FPU and CTRLAP and GPIOTE1 did not match 100%, so kept them as they were.